server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -172,6 +161,10 @@ public final String simpleName() {
      *  Both {@code this} and {@code mergeWith} will be left unmodified. */
     public abstract Mapper merge(Mapper mergeWith);
 
+    public void collectPerFieldResources(PerFieldResourceCollector collector) {
```

Comment:
If I'm understanding correctly, it's still possible to set a similarity on a non-text field, but we now ignore it. Although I agree that similarities aren't nearly as useful for non-text fields, this behavior seems trappy and it's technically a breaking change.

My vote would be to maintain the previous behavior where similarities on types like `keyword` were respected. If we want to disallow setting similarities on non-text types, we could open a dedicated issue to discuss it.